File: ruby-pattern/iterator/iterator.rb

class Iterator

  def initialize(array)
    @array = array
    @index = 0
  end

  def has_next
    # compare with index
    return (@index + 1) <= @array.count
  end

  def next
    value = @array[@index]
    @index = @index + 1
    value
  end

end

File: swift-pattern/Iterator.playground/Sources/Iterator.swift
public class Iterator {

    var index:Int = 0
    var array:[Any] = []

    public init(array:[Any]){
        self.array = array
    }

    public func hasNext() -> Bool {
        return (index < array.count)
    }

    public func next()->Any {
        let value:Any = array[index]
        index = index+1
        return value
    }

}

